src/main/java/org/example/service/CommentService.java

```diff
@@ -0,0 +1,40 @@
+package org.example.service;
+
+import org.example.entity.Comment;
+import org.example.entity.User;
+import org.example.repository.CommentRepository;
+import org.example.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class CommentService {
+
+    @Autowired
+    private CommentRepository commentRepository;
+
+    @Autowired
+    private UserRepository userRepository;
```

Comment:
You need to use repositories only in the cooresponding services (eg. User Service -> UserRepository, etc.)
Use UserService here.
And rework in other places too